tests/conversion_02.cc

```diff
@@ -0,0 +1,68 @@
+
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the SampleFlow authors.
+//
+// This file is part of the SampleFlow library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Check the conversion filter
+
+
+#include <iostream>
+
+#include <sampleflow/consumers/stream_output.h>
+#include <sampleflow/filters/conversion.h>
+#include <sampleflow/producers/range.h>
+
+/**
+ * Example conversion function; converts a string to an integer
+ * by getting the sum of the ASCII codes of each character.
+ */
+int string_ascii_sum(std::string sample)
+{
+    int sum = 0;
+    for(char& c : sample)
```

Comment:
Good practice since you're not going to change `c`:
```suggestion
    for(const char& c : sample)
```

Reply:
https://en.cppreference.com/w/cpp/language/range-for